pkg/controllers/applicationset_controller.go

```diff
@@ -67,6 +70,23 @@ func (r *ApplicationSetReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	// Log a warning if there are unrecognized generators
```

Comment:
Nitpick: I think this block can be moved into its own function (eg `checkInvalidGenerators`), which can then be called from here.

Reply:
Done